### PR TITLE
switch to UPF specific timer wheel implementation

### PR DESCRIPTION
**Pull request by RoadRunnr**

### CODE REVIEW

**Review by ivan4th — CHANGES_REQUESTED** [review]
One quite small change might be needed, also there's typo in one of the commit descriptions ("stoped"):

> Handles must now be allocated before they can be started and they are not
longer automatically freed when stoped. Reusing handles for restarting
or changing timer is now possible.

should be

> Handles must now be allocated before they can be started and are no
longer automatically freed when stopped. Reusing handles for restarting
or changing the timer is now possible.

File: upf/upf_pfcp_server.c [hunk 1]
```diff
@@ -1368,7 +1308,8 @@ static uword
 
       /* run the timing wheel first, to that the internal base for new and updated timers
        * is set to now */
-      upf_pfcp_server_expire_timers ();
+      psm->expired =
+	TW (tw_timer_expire_timers_vec) (&psm->timer, psm->now, psm->expired);
```
[the remark this answers is not in the document]
Comment (RoadRunnr) [reply upf/upf_pfcp_server.c]: done

**Review by ivan4th — APPROVED** [review]

**Review by ivan4th — CHANGES_REQUESTED** [review]
Almost approved, but there's a checkstyle error
```
Files upf/upf_pfcp_server.c and upf/upf_pfcp_server.c.out2 differ

Checkstyle failed for upf/upf_pfcp_server.c.
Run indent (twice!) as shown to fix the problem:
indent upf/upf_pfcp_server.c
indent upf/upf_pfcp_server.c
*******************************************************************
* VPP CHECKSTYLE FAILED
* CONSULT FAILURE LOG ABOVE
* NOTE: Running 'hack/checkstyle.sh --fix' *MAY* fix the issue
*******************************************************************
make: *** [Makefile:29: checkstyle] Error 1
```

**Review by ivan4th — APPROVED** [review]
LGTM

**Pull request merged**